### PR TITLE
[SPARK-6551][PYSPARK] Incorrect aggregate results if seqOp(...) mutates its first argument

### PR DESCRIPTION
**Pull request by megatron-me-uk**
Implement a test for SPARK-6551.

### CODE REVIEW

**Comment by srowen** [comment]
(Can you add a little more to the title, like what the underlying JIRA is about?)

**Comment by megatron-me-uk** [comment]
I believe that this will now resolve the issues reported in SPARK-6551 and test for regression.

File: python/pyspark/rdd.py [hunk 1]
```diff
@@ -898,7 +899,7 @@ def func(iterator):
         # zeroValue provided to each partition is unique from the one provided
```
Comment (JoshRosen) [inline python/pyspark/rdd.py]: This comment needs to be updated.

Comment (JoshRosen) [inline python/pyspark/rdd.py]: Or removed.

**Comment by davies** [comment]
This bug is fixed by #7378 (merged into 1.4 and 1.5) , Could you send a patch against 1.3 branch (remove the unrelated changes)?

**Comment by megatron-me-uk** [comment]
#7378 has been merged into master and fails half of the tests that I have added in this pull request. The returned result is correct however the `zeroValue` value supplied is changed. Is it desirable to change the `zeroValue`that is supplied to rdd.aggregate?

**Comment by davies** [comment]
@megatron-me-uk I think it's reasonable to change the `zeroValue` once you provide a mutable function, for example, the builtin function `reduce` also works in this way.

If we change to reply on `deepcopy`, it will not work in some cases (for example, object from C extension).

**Comment by megatron-me-uk** [comment]
OK good point, I guess I will close this pull request then. I think the best way to resole this jira issue is to backport #7378 or just recommend an upgrade to 1.4.

**Pull request closed without merging**